[CI] XPackRestIT test {p0=text_structure/find_structure/Test NDJSON file structure analysis without overrides} timing out

This test sometimes times out now.  The problem seems to have been introduced within the last 24 hours.  Since it's affecting the 7.13 branch I suspect #74581 is involved.

**Build scan:**
https://gradle-enterprise.elastic.co/s/qgmzvmu6uhxuu/tests/:x-pack:plugin:yamlRestTest/org.elasticsearch.xpack.test.rest.XPackRestIT/test%20%7Bp0=text_structure%2Ffind_structure%2FTest%20NDJSON%20file%20structure%20analysis%20without%20overrides%7D

**Reproduction line:**
`./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=text_structure/find_structure/Test NDJSON file structure analysis without overrides}" -Dtests.seed=FFB537FF4CDA313A -Dtests.locale=nl -Dtests.timezone=Asia/Katmandu -Druntime.java=8`

**Applicable branches:**
master, 7.x, 7.14, 7.13

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.test.rest.XPackRestIT&tests.test=test%20%7Bp0%3Dtext_structure/find_structure/Test%20NDJSON%20file%20structure%20analysis%20without%20overrides%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [text_structure/find_structure:6]: expected [2xx] status code but api [text_structure.find_structure] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"timeout_exception","reason":"Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]","stack_trace":"ElasticsearchTimeoutException[Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.check(TimeoutChecker.java:89)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.grokCaptures(TimeoutChecker.java:110)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.checkCandidate(TimestampFormatFinder.java:526)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.addSample(TimestampFormatFinder.java:589)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.findCandidates(TextStructureUtils.java:227)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.guessTimestampField(TextStructureUtils.java:120)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinder.makeNdJsonTextStructureFinder(NdJsonTextStructureFinder.java:63)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinderFactory.createFromSample(NdJsonTextStructureFinderFactory.java:86)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.makeBestStructureFinder(TextStructureFinderManager.java:611)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:401)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n\tSuppressed: ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\t\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\t\t... 7 more\n","suppressed":[{"type":"exception","reason":"Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n","stack_trace":"ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}]}],"type":"timeout_exception","reason":"Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]","stack_trace":"ElasticsearchTimeoutException[Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.check(TimeoutChecker.java:89)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.grokCaptures(TimeoutChecker.java:110)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.checkCandidate(TimestampFormatFinder.java:526)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.addSample(TimestampFormatFinder.java:589)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.findCandidates(TextStructureUtils.java:227)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.guessTimestampField(TextStructureUtils.java:120)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinder.makeNdJsonTextStructureFinder(NdJsonTextStructureFinder.java:63)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinderFactory.createFromSample(NdJsonTextStructureFinderFactory.java:86)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.makeBestStructureFinder(TextStructureFinderManager.java:611)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:401)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n\tSuppressed: ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\t\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\t\t... 7 more\n","suppressed":[{"type":"exception","reason":"Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n","stack_trace":"ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}]},"status":500}]

  at __randomizedtesting.SeedInfo.seed([FFB537FF4CDA313A:77E10825E2265CC2]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:451)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:428)
  at sun.reflect.GeneratedMethodAccessor15.invoke(null:-1)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [text_structure.find_structure] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"timeout_exception","reason":"Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]","stack_trace":"ElasticsearchTimeoutException[Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.check(TimeoutChecker.java:89)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.grokCaptures(TimeoutChecker.java:110)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.checkCandidate(TimestampFormatFinder.java:526)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.addSample(TimestampFormatFinder.java:589)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.findCandidates(TextStructureUtils.java:227)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.guessTimestampField(TextStructureUtils.java:120)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinder.makeNdJsonTextStructureFinder(NdJsonTextStructureFinder.java:63)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinderFactory.createFromSample(NdJsonTextStructureFinderFactory.java:86)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.makeBestStructureFinder(TextStructureFinderManager.java:611)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:401)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n\tSuppressed: ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\t\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\t\t... 7 more\n","suppressed":[{"type":"exception","reason":"Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n","stack_trace":"ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}]}],"type":"timeout_exception","reason":"Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]","stack_trace":"ElasticsearchTimeoutException[Aborting structure analysis during [timestamp format determination] as it has taken longer than the timeout of [10s]]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.check(TimeoutChecker.java:89)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimeoutChecker.grokCaptures(TimeoutChecker.java:110)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.checkCandidate(TimestampFormatFinder.java:526)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TimestampFormatFinder.addSample(TimestampFormatFinder.java:589)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.findCandidates(TextStructureUtils.java:227)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureUtils.guessTimestampField(TextStructureUtils.java:120)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinder.makeNdJsonTextStructureFinder(NdJsonTextStructureFinder.java:63)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.NdJsonTextStructureFinderFactory.createFromSample(NdJsonTextStructureFinderFactory.java:86)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.makeBestStructureFinder(TextStructureFinderManager.java:611)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:401)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n\tSuppressed: ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\t\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\t\t... 7 more\n","suppressed":[{"type":"exception","reason":"Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n","stack_trace":"ElasticsearchException[Explanation so far:\n[Using character encoding [UTF-8], which matched the input with [15%] confidence - first [8kB] of input was pure ASCII]\n[Deciding sample is newline delimited NDJSON]\n]\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:415)\n\tat org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager.findTextStructure(TextStructureFinderManager.java:340)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.buildTextStructureResponse(TransportFindStructureAction.java:54)\n\tat org.elasticsearch.xpack.textstructure.transport.TransportFindStructureAction.lambda$doExecute$0(TransportFindStructureAction.java:42)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}]},"status":500}]

    at org.junit.Assert.fail(Assert.java:88)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:354)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:444)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:428)
    at sun.reflect.GeneratedMethodAccessor15.invoke(null:-1)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
    at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
    at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
    at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
    at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
    at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
    at java.lang.Thread.run(Thread.java:748)

```

--- a/libs/grok/src/main/java/org/elasticsearch/grok/Grok.java
+++ b/libs/grok/src/main/java/org/elasticsearch/grok/Grok.java
@@ -79,11 +79,7 @@ public final class Grok {
         this.namedCaptures = namedCaptures;
         this.matcherWatchdog = matcherWatchdog;
 
-        for (Map.Entry<String, String> entry : patternBank.entrySet()) {
-            String name = entry.getKey();
-            String pattern = entry.getValue();
-            forbidCircularReferences(name, new ArrayList<>(), pattern);
-        }
+        forbidCircularReferences();
 
         String expression = toRegex(grokPattern);
         byte[] expressionBytes = expression.getBytes(StandardCharsets.UTF_8);
@@ -104,7 +100,8 @@ public final class Grok {
      * a reference to another named pattern. This method will navigate to all these named patterns and
      * check for a circular reference.
      */
-    private void forbidCircularReferences(String patternName, List<String> path, String pattern) {
+    private void forbidCircularReferences() {
+
         // first ensure that the pattern bank contains no simple circular references (i.e., any pattern
         // containing an immediate reference to itself) as those can cause the remainder of this algorithm
         // to recurse infinitely
@@ -114,8 +111,12 @@ public final class Grok {
             }
         }
 
-        // next recursively check any other pattern names referenced in the pattern
-        innerForbidCircularReferences(patternName, path, pattern);
+        // next, recursively check any other pattern names referenced in each pattern
+        for (Map.Entry<String, String> entry : patternBank.entrySet()) {
+            String name = entry.getKey();
+            String pattern = entry.getValue();
+            innerForbidCircularReferences(name, new ArrayList<>(), pattern);
+        }
     }
 
     private void innerForbidCircularReferences(String patternName, List<String> path, String pattern) {
@@ -151,7 +152,7 @@ public final class Grok {
             }
             String otherPatternName = pattern.substring(begin, end);
             path.add(otherPatternName);
-            forbidCircularReferences(patternName, path, patternBank.get(otherPatternName));
+            innerForbidCircularReferences(patternName, path, patternBank.get(otherPatternName));
         }
     }
 

